### PR TITLE
Feature/handle stale transactions

### LINKED ISSUE #369: Implement transaction fetching for non-mined ones
Currently while indexing some transactions don't have their whole dataset when querying via GraphQL. We need to ensure that these transactions are filled to have a proper indexed chain state.

We need to rework the pbft module so that it:
-  Registers a `${prefix}_stale_transactions` queue in Redis.
- The `pbft.consumer.ts` checks for all "stale" transactions after saving the block and puts their hash in the queue.

 We need to implement a new module in the `explorer-indexer` service that:

- Consumes these hashes and pulls the transactions via GQL.
- Updates the transaction in-DB or puts the hash back in the queue.

### PR DESCRIPTION
**Pull request by VargaElod23**
Closes #369 .

### CODE REVIEW

File: services/explorer-indexer/src/modules/health/producerHealthIndicator.ts [hunk 1]
```diff
@@ -57,6 +65,12 @@ export class ProducerHealthIndicator extends HealthIndicator {
         }
         break;
       }
+      case 'queue_transactions': {
+        if (this.dagService) {
```
Comment (rattrap) [inline services/explorer-indexer/src/modules/health/producerHealthIndicator.ts:69]: `this.txService`

File: services/explorer-indexer/src/modules/health/syncerHealthIndicator.ts [hunk 2]
```diff
@@ -55,6 +63,12 @@ export class SyncerHealthIndicator extends HealthIndicator {
         }
         break;
       }
+      case 'queue_transactions': {
+        if (this.dagService) {
```
Comment (rattrap) [inline services/explorer-indexer/src/modules/health/syncerHealthIndicator.ts:67]: `this.txService`

File: services/explorer-indexer/src/modules/historical-sync/historical.syncer.module.ts [hunk 3]
```diff
@@ -9,6 +9,7 @@ import general from '../../config/general';
 import { ConnectorsModule } from '../connectors';
 import { BullModule } from '@nestjs/bull';
 import * as dotenv from 'dotenv';
+import { Queues } from 'src/types';
```
Comment (rattrap) [inline services/explorer-indexer/src/modules/historical-sync/historical.syncer.module.ts:12]: Relative imports

File: services/explorer-indexer/src/modules/pbft/pbft.consumer.ts [hunk 4]
```diff
@@ -107,4 +121,44 @@ export class PbftConsumer implements OnModuleInit {
       await job.progress(100);
     }
   }
+
+  async handleStaleTransactions(pbft: PbftEntity, syncType: SyncTypes) {
+    if (!pbft.transactions && pbft.transactions?.length === 0) return;
```
Comment (rattrap) [inline services/explorer-indexer/src/modules/pbft/pbft.consumer.ts:126]: I think you wanted to have an or here

File: services/explorer-indexer/src/modules/pbft/pbft.consumer.ts [hunk 5]
```diff
@@ -107,4 +121,44 @@ export class PbftConsumer implements OnModuleInit {
...
+      `${pbft.number} has ${staleCount} stale transactions. Iterating: `
+    );
+    if (staleCount) {
+      for (const transaction of pbft.transactions) {
+        try {
+          if (
+            transaction &&
+            transaction.hash &&
+            !transaction.status &&
+            !transaction.value
```
Comment (rattrap) [inline services/explorer-indexer/src/modules/pbft/pbft.consumer.ts:139]: We already check these

File: services/explorer-indexer/src/modules/pbft/pbft.consumer.ts [hunk 6]
```diff
@@ -107,4 +121,44 @@ export class PbftConsumer implements OnModuleInit {
...
+              this.logger.log(
+                `Pushed stale transaction ${transaction.hash} into ${this.txQueue.name} queue`
+              );
+            }
+          }
+        } catch (error) {
+          this.logger.error(
+            `Pushing into ${this.txQueue.name} ran into an error: ${error}`
+          );
+          await this.txQueue.add(QueueJobs.STALE_TRANSACTIONS, {
```
Comment (rattrap) [inline services/explorer-indexer/src/modules/pbft/pbft.consumer.ts:156]: If it failed in the try, why wouldn't it fail in the catch?

**Review by rattrap — CHANGES_REQUESTED** [review]

**Review by Malak67 — APPROVED** [review]

**Pull request merged**